universal-application-tool-0.0.1/app/services/question/ReadOnlyQuestionServiceImpl.java

```diff
@@ -44,20 +48,28 @@ public ReadOnlyQuestionServiceImpl(ImmutableList<QuestionDefinition> questions)
                   scalarMap.put(entry.getKey(), entry.getValue());
                 });
       }
+      upToDateBuilder.add(
+          qds.stream().max(Comparator.comparing(QuestionDefinition::getVersion)).get());
     }
     for (QuestionDefinition qd : questions) {
       questionIdMap.put(qd.getId(), qd);
     }
     questionsById = questionIdMap.build();
     questionsByPath = questionPathMap.build();
     scalars = scalarMap.build();
+    upToDateQuestions = upToDateBuilder.build();
   }
 
   @Override
   public ImmutableList<QuestionDefinition> getAllQuestions() {
     return questionsById.values().asList();
   }
 
+  @Override
+  public ImmutableList<QuestionDefinition> getUpToDateQuestions() {
+    return upToDateQuestions.asList();
+  }
+
   @Override
   public ImmutableList<RepeaterQuestionDefinition> getRepeaterQuestions() {
     return getAllQuestions().stream()
```

Comment:
Issue (blocking): update `getRepeaterQuestions()` to use `getUpToDateQuestions()` internally

Reply:
Done - changed name to make clear, and added `getAll` equivalent.